pkg/genlib/generator_with_custom_template_test.go

```diff
@@ -228,10 +218,26 @@ func test_CardinalityTWithCustomTemplate[T any](t *testing.T, ty string) {
 	// It's cardinality per mille, so a bit confusing :shrug:
 	for cardinality := 1000; cardinality >= 10; cardinality /= 10 {
 
+		cardinalityDenominator := 1000
+		cardinalityNumerator := cardinality
+		cardinalityModule := cardinalityDenominator % cardinality
+		if cardinalityModule == 0 {
+			cardinalityNumerator = 1
+			cardinalityDenominator /= cardinality
+		}
+
+		rangeTrailing := ""
+		if ty == FieldTypeFloat {
+			rangeTrailing = "."
+		}
+
+		rangeMin := rand.Intn(100)
+		rangeMax := rand.Intn(10000-rangeMin) + rangeMin
```

Comment:
as you know, I prefer to have some fuzziness in tests, to catch potential hidden bugs ;)

Reply:
I'm not sure this test is actually behaving like that. Without updating the `seed` the generated values will not change. ([Seed docs](https://pkg.go.dev/math/rand#Seed), it behaves like always using `Seed(1)`)

See for a playground: https://go.dev/play/p/vCLM9roYzuw

From my understanding without updating the seed (I'm not sure if this happens somewhere, but searching for `Seed` or `rand.Seed` in this repo given 0 results) this code behave like using a constant value.

Reply:
> From my understanding without updating the seed (I'm not sure if this happens somewhere, but searching for `Seed` or `rand.Seed` in this repo given 0 results) this code behave like using a constant value.

yes, I don't update the seed, indeed. but that's not needed: every call to `rand.Intn` does not generate a constant value, rather, multiple calls with the same seed generate always the same sequence: 
> Seed values that have the same remainder when divided by 2³¹-1 generate the same pseudo-random sequence

I suppose that once the sequence is completely generated it will start over. I have to look how long is the sequence (int64 size?) but we are far from filling it :)

so basically we test always with the same sequence, indeed, that's why I mentioned that "we can also have a fixed list of ranges for every cardinality" :)

or we can use the current unix timestamp as seed, reducing the number of identical sequences between each tests run